### PR TITLE
Provisioning: provisioning playlists

### LINKED ISSUE #11842: [Feature request] playlist provisioning
I'm using the feature of provisioning dashboards and datasources. These are good for deploying to kubernetes. I want to use provisioning playlist too 😄

### PR DESCRIPTION
**Pull request by olivierlemasle**
**What this PR does / why we need it**:
This PR enables provisioning of playlists, in a similar way to provisioning of dashboards, datasources and notifiers.

When Grafana is deployed using GitOps, this is required to add playlists easily along with dashboards.

**Which issue(s) this PR fixes**:

Fixes #11842

**Special notes for your reviewer**:
In order to have a unique user-facing ID for playlists, I added a new field `uid` to playlists.

Perhaps I should also add new APIs using `uid`? I'd like to have the opinion of a reviewer/maintainer before adding APIs commands and documentation...
I also thought about a new URL to get playlist display, such as http://grafana:3000/playlists/play/uid/myuid?kiosk. Could that be useful?

### CODE REVIEW

**Comment by olivierlemasle** [comment]
Thank you @marefr for your answer. Do you have any idea when will this be reviewed?

Afterwards I'd like to tackle #12119, also related to provisioning.

**Comment by olivierlemasle** [comment]
Pinging to remove "stale" label.
This PR and its review depend on the "additional thinking" related to provisioning discussed in https://github.com/grafana/grafana/issues/12119#issuecomment-553002801.

**Comment by marefr** [comment]
@olivierlemasle we've done some additional thinking now and have come to the conclusion that we cannot currently accept this. We're still not sure we want to continue adding additional provisioning support in the same way as already done for dashboards, data sources and alert notification channels.  

However, adding `uid` for playlists and other types/entities (you mentioned organizations) we definitely feel is a good thing and a proper first step to make Grafana more future proof and easier to work with from an HTTP API perspective, and in addition if you have multiple Grafana instances/environments. Have created an epic for this #20830.

I don't know If you're still interested in contributing `uid` support for playlists, but if you are please let me know and I can create a specific issue for that one. I'll close this for now, but doesn't mean we cannot reopen it later.

Again, thank you for your contribution and really sorry it took long time for us come back with a response.

**Comment by olivierlemasle** [comment]
Thanks @marefr for your reply.

This is perhaps not the place to discuss it, but what is your team's current position regarding provisioning?

I recognize that current provisioning system is not perfect, especially regarding deletion: provisioning mixes a _declarative_ process (e.g. ensure datasources exist and are configured) with an _imperative_ process (e.g. `deleteDatasources` to delete existing datasources) - cf #12875 and #15716.

However, provisioning only a subset of Grafana's features has a limited interest and do not allow a _GitOps_ way of managing Grafana.

Using GitOps (e.g. with Grafana deployed as part of https://github.com/coreos/kube-prometheus, or using a CI pipeline to deploy and configure Grafana) requires everything to be "provisionable", from organizations (#12119) to playlists (#11842) or users (#19848). Please note that these three feature requests are currently still opened and have many votes from the community (as of today, respectively 14, 26 and 3 votes). A clarification of provisioning's future would be much appreciated!

**Comment by marefr** [comment]
@olivierlemasle Sorry if I was unclear. We're not saying that we don't want to support provisioning of everything, we're just saying that right now adding additional provisioning support in same way as existing dashboards, data sources and alert notification channels adds quite a big impact on core implementation  What's lacking is more proper API's internally to make provisioning a first class citizen in the code base, but without affecting any bits in the core (domain). Provisioning should be a client just just like the Grafana HTTP API.

In summary, GitOps way of managing Grafana is definitely something we want to support. Then it can be discussed whether provisioning of users is something we eventually would support given that there's already support for external authentication providers like OAuth, LDAP and SAML. But as always everything is up for discussion and we try to listen on our community.

I totally agree that deletion in provisioning currently is a bad experience and something we're planning to address.

**Comment by lanmarti** [comment]
@marefr 
>Provisioning should be a client just just like the Grafana HTTP API
Is there an issue tracking this, or how would one stay in the loop about progress on this client?
